### ISSUE TITLE
(Android) APK install not working for Android <7

### ISSUE THREAD

**Issue by luandro** [digidem/mapeo-mobile#582]
The Java [code we're using](https://github.com/digidem/mapeo-mobile/blob/share-apk-p2p/android/app/src/main/java/com/mapeo/ApkInstallerModule.java) for installing apks throws error on my Android 6 device:
```
W PackageInstaller: Parse error when parsing manifest. Discontinuing installation
```
This could be a permission problem for Android, as on Android 6 they are not asked for:
![mapeo_permissions](https://user-images.githubusercontent.com/693728/117206892-c3351600-adc9-11eb-9047-7a625c3d62c0.jpg)

But there's no way to give permission through the settings and it's confusing if the has the permissions or not.

### Paths foward:

- Check [original package](https://github.com/nodece/react-native-apk-installer-n/blob/master/android/src/main/java/com/nodece/apkinstallern/RNApkInstallerNModule.java) for any updates on the code
- Reactivate [similar issue](https://github.com/nodece/react-native-apk-installer-n/issues/3) on original repo
- Compare code from [react-native-apk-installer](https://github.com/null--/react-native-apk-installer/blob/master/android/src/main/java/com/cnull/apkinstaller/ApkInstallerModule.java) which I remember worked on my phone many years ago

Other ideas are welcome!

**Comment by luandro** [comment]
Did some investigation on [dat-installer](https://github.com/staltz/dat-installer) code to check if any additional permissions were asked for anywhere. But didn't find anything different from what we're doing. It seems there's no need to explicitily ask for permission for installing apks.

Found on a Stackoverflow that `<uses-permission android:name="android.permission.INSTALL_PACKAGES" />` might be needed on the AndroidManifest, despite dat-installer not using it. But adding it made no difference.

**Comment by hackergrrl** [comment]
On 05/06  14:33, luandro wrote:
> Did some investigation on [dat-installer](https://github.com/staltz/dat-installer) code to check if any additional permissions were asked for anywhere. But didn't find anything different from what we're doing. It seems there's no need to explicitily ask for permission for installing apks.

Does datinstaller work ok for you? Or does it have the same problem on
your phone?

**Comment by luandro** [comment]
It worked in the past, but trying it out now it seems I'm having a problem with Dat, which isn't downloading the apps. Tried with `dat://manyverse-latest.hashbase.io` and `dat://dat-installer-latest.hashbase.io` which are always beeing seeded by hashbase, and also tried seeding an app myself. But I'm always stuck on `searching`....

Even without downloading the app, by clicking on the Dat address, a windows with an install options shows up. And by clicking install it obviously fails, but interesting enough, with the same error as in Mapeo:
```
W PackageInstaller: Parse error when parsing manifest. Discontinuing installation
```

**Comment by luandro** [comment]
Fixed by #586
